app/permissions_migrations.go

```diff
@@ -0,0 +1,157 @@
+// Copyright (c) 2018-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package app
+
+import "github.com/mattermost/mattermost-server/model"
+
+type permissionTransformation struct {
+	On     func(map[string]bool) bool
+	Add    []string
+	Remove []string
+}
+type permissionsMap []permissionTransformation
+
+const (
+	MIGRATION_KEY_EMOJI_PERMISSIONS_SPLIT   = "emoji_permissions_split"
+	MIGRATION_KEY_WEBHOOK_PERMISSIONS_SPLIT = "webhook_permissions_split"
+
+	PERMISSION_MANAGE_EMOJIS                   = "manage_emojis"
+	PERMISSION_MANAGE_OTHERS_EMOJIS            = "manage_others_emojis"
+	PERMISSION_CREATE_EMOJIS                   = "create_emojis"
+	PERMISSION_DELETE_EMOJIS                   = "delete_emojis"
+	PERMISSION_DELETE_OTHERS_EMOJIS            = "delete_others_emojis"
+	PERMISSION_MANAGE_WEBHOOKS                 = "manage_webhooks"
+	PERMISSION_MANAGE_OTHERS_WEBHOOKS          = "manage_others_webhooks"
+	PERMISSION_MANAGE_INCOMING_WEBHOOKS        = "manage_incoming_webhooks"
+	PERMISSION_MANAGE_OTHERS_INCOMING_WEBHOOKS = "manage_others_incoming_webhooks"
+	PERMISSION_MANAGE_OUTGOING_WEBHOOKS        = "manage_outgoing_webhooks"
+	PERMISSION_MANAGE_OTHERS_OUTGOING_WEBHOOKS = "manage_others_outgoing_webhooks"
+)
+
+func permissionExists(permission string) func(map[string]bool) bool {
+	return func(permissions map[string]bool) bool {
+		val, ok := permissions[permission]
+		return ok && val
+	}
+}
+
+func permissionNotExists(permission string) func(map[string]bool) bool {
+	return func(permissions map[string]bool) bool {
+		val, ok := permissions[permission]
+		return !(ok && val)
+	}
+}
```

Comment:
Because is not a function that returns a boolean, is a function that returns a function that returns a boolean. The idea to have a "builder" of functions that checks permissions.